osidb/tasks.py

```diff
@@ -0,0 +1,99 @@
+from celery import states as celery_states
+from celery.utils import log
+from django.conf import settings
+from django.core.mail import EmailMessage
+from django.db.models import Count
+from django.utils import timezone
+from django_celery_results.models import TaskResult
+
+from config.celery import app
+
+logger = log.get_task_logger(__name__)
+
+
+def get_last_success_for_task(task_name):
+    """Return the timestamp of the last successful task so we can fetch updates since that time.
+
+    For extra measure, the last success timestamp is offset by 30 minutes to overlap. If no record
+    of a job that succeeded exists in our results DB, return a refresh timestamp of 3 days ago.
+    If that still misses stuff, it indicates a longer outage and updates should be scheduled
+    manually.
+    """
+    # Specifically query for jobs without any task kwargs so prevent refreshing only after a
+    # successful manually triggered task for a particular resource.
+    last_success = (
+        TaskResult.objects.filter(
+            task_name=task_name, task_kwargs='"{}"', status="SUCCESS"
+        )
+        .order_by("-date_done")
+        .values_list("date_done", flat=True)
+        .first()
+    )
+    return (
+        last_success - timezone.timedelta(minutes=30)
+        if last_success
+        else timezone.now() - timezone.timedelta(days=3)
+    )
+
+
+@app.task(autoretry_for=(Exception,), retry_backoff=900, retry_jitter=False)
+def email_failed_tasks():
+    """Send email about failed Celery tasks within past 24 hours to OSIDB admins (mailing list)
+    If it failed to send, try again after 15 minutes, then 30 minutes, then give up"""
+    failed_tasks_threshold = get_last_success_for_task("osidb.tasks.email_failed_tasks")
+    failed_tasks = TaskResult.objects.filter(
+        status__in=(celery_states.FAILURE, celery_states.RETRY),
```

Comment:
Could you remove the `RETRY` state from the filters? We generate a lot of those because a few collectors can't run in parallel, furthermore even for the cases where it'd make sense it'll either eventually succeed or fail and will be caught then.